src/main/java/part2/cache/TypedEmployeeCachedStorage.java

```diff
@@ -19,8 +24,48 @@ public TypedEmployeeCachedStorage(CachingDataStorage<String, data.Employee> empl
     }
 
     @Override
-    public OutdatableResult<Employee> getOutdatable(String key) {
+    public OutdatableResult<data.typed.Employee> getOutdatable(String key) {
         // TODO note that you don't know timeouts for different storage. And timeouts can be different.
-        throw new UnsupportedOperationException();
+        final OutdatableResult<data.Employee> notTypedOutdatable = employeeStorage.getOutdatable(key);
+        final CompletableFuture<Employee> typedEmployeeCf = new CompletableFuture<>();
+        final CompletableFuture<Void> outdatedCf = new CompletableFuture<>();
+
+        notTypedOutdatable.getResult().whenComplete((emp, ex) -> {
+            if (ex != null) {
+                typedEmployeeCf.completeExceptionally(ex);

```

Comment:
It' s provided by adding `notTypedOutdatable.getOutdated` to list of outdated signals and further calling `CompletedFuture.anyOf(outdatedSignals)`